src/util/clipSketchToSubregion.ts

```diff
@@ -0,0 +1,227 @@
+import {
+  InternalVectorDatasource,
+  Polygon,
+  Sketch,
+  SketchCollection,
+  Feature,
+  MultiPolygon,
+  toSketchArray,
+} from "@seasketch/geoprocessing/client-core";
+import { fgbFetchAll } from "@seasketch/geoprocessing/dataproviders";
+import { featureCollection } from "@turf/helpers";
+import bbox from "@turf/bbox";
+import project from "../../project";
+import geographies from "../../project/geographies.json";
+import {
+  clipMultiMerge,
+  getFlatGeobufFilename,
+} from "@seasketch/geoprocessing";
+import simplify from "@turf/simplify";
+
+export interface ExtraParams {
+  /** Optional ID(s) of geographies to operate on. **/
+  geographies?: string[];
+}
```

Comment:
I would suggest moving this into a top level `types.ts` file that all files import from.  Consider doing the same with other types that are reused across multiple files and more like building blocks

Reply:
I would put `types.ts` up one level higher than you have it now, out of the util folder.  Projects in the wild I see tend to make them a first class thing, and when I see a types file, I know I can go there learn about how data is structured for the project.  The gp library has its own types [folder](https://github.com/seasketch/geoprocessing/tree/dev/packages/geoprocessing/src/types), but you can simplify it to a single types.ts file in this case, until you outgrow it.

Reply:
This sounds good, I'll move it up a level.